Sources/Datadog/Logging/RemoteLogger.swift

```diff
@@ -55,41 +53,36 @@ internal final class RemoteLogger: LoggerProtocol {
         self.core = core
         self.configuration = configuration
         self.dateProvider = dateProvider
-        self.queue = DispatchQueue(
-            label: "com.datadoghq.logger-\(configuration.loggerName)",
-            target: .global(qos: .userInteractive)
-        )
-
         self.rumContextIntegration = rumContextIntegration
         self.activeSpanIntegration = activeSpanIntegration
     }
 
     // MARK: - Attributes
 
     func addAttribute(forKey key: AttributeKey, value: AttributeValue) {
-        queue.async { self.unsafeAttributes[key] = value }
```

Comment:
wouldn't the `queue.sync` solve the issue described in the description?

Reply:
The issue comes from the fact that we capture attributes and tags on the context queue when writing a log entry, which introduce the inconsistency like in this sequence:
```swift
logger.add(tag: "tag1")
logger.info("info message 1") // This log entry won't include tag1 because the EWC is async
logger.remove(tag: "tag1")
```

Using `sync` when writing attributes or tags won't solve the issue. We need to get the tags and attributes before invoking the `. eventWriteContext` and we need to invoke `.eventWriteContext` on the caller thread so the Core Context is also consistent.